### PR TITLE
Publish wheel events

### PR DESCRIPTION
**Pull request by keller-mark**
## Description

> What was changed in this pull request?

This pull request adds support for a `.on('wheel')` callback to enable the wheel event to be published. It also adds an API function `.setWheelCallbackKeydownCondition(keyCode)`, which if used, ensures that the wheel callback is only called when the specified keyboard key has been pressed down.

> Why is it necessary?

This will enable us to catch the wheel event in the cistrome wrapper to support "zooming" along the y axis, which we will try to implement by listening for the wheel event and updating the `selectRows` array. (Though we are still not sure how smooth this approach will be)

Fixes #___

## Checklist

- [ ] Set proper GitHub labels (e.g. v1.6+, ignore if you don't know)
- [x] Unit tests added or updated
- [x] Documentation added or updated
- [ ] Example(s) added or updated
- [ ] Update schema.json if there are changes to the viewconf JSON structure format
- [ ] Screenshot for visual changes (e.g. new tracks or UI changes)
- [x] Updated CHANGELOG.md

### CODE REVIEW

**Comment by keller-mark** [comment]
Your solution is much nicer, I have adopted that, thanks.

File: app/scripts/HiGlassComponent.js [hunk 1]
```diff
@@ -820,12 +825,40 @@ class HiGlassComponent extends React.Component {
...
+
   keyDownHandler(event) {
     if (this.props.options.rangeSelectionOnAlt && event.key === 'Alt') {
       this.setState({
         mouseTool: MOUSE_TOOL_SELECT
       });
     }
+
+    if (this.isKeyDownWheelConditionSatisfied(event.key)) {
+      this.shouldPublishWheelEvent = true;
```
Comment (pkerpedjiev) [inline app/scripts/HiGlassComponent.js]: What if you register two wheel event handlers? Won't this cause cross-talk?

**Comment by keller-mark** [comment]
I am just going to remove the whole keydown condition thing. We should be able to listen for those events from outside higlass and then set / remove the wheel handler accordingly.

**Comment by keller-mark** [comment]
@pkerpedjiev I removed the keydown parameter and now I just check if the number of wheel callbacks is greater than zero, where I increment / decrement with each call to .on / .off

https://github.com/higlass/higlass/pull/876/files#diff-e80a6b2d4c494c27266c4c575997a52eR891

Is that alright or is there a better approach?

File: app/scripts/HiGlassComponent.js [hunk 2]
```diff
@@ -4288,6 +4291,52 @@ class HiGlassComponent extends React.Component {
...
+            : relTrackPos[0],
+        relTrackY:
+          hoveredTrack && hoveredTrack.flipText
+            ? relTrackPos[0]
+            : relTrackPos[1],
+        track: hoveredTrack,
+        origEvt: nativeEvent,
+        sourceUid: this.uid,
+        hoveredTracks,
+        noHoveredTracks: hoveredTracks.length === 0
```
Comment (pkerpedjiev) [inline app/scripts/HiGlassComponent.js]: Why is this necessary? Can't the use just do the length check themselves?

File: app/scripts/api.js [hunk 3]
```diff
@@ -682,6 +683,11 @@ const createApi = function api(context, pubSub) {
             apiPubSub.unsubscribe('mouseMoveZoom', callback);
             break;
 
+          case 'wheel':
+            self.numWheelCallbacks -= 1;
```
Comment (pkerpedjiev) [inline app/scripts/api.js]: I wonder if there's a way to just count the number of listeners in apiPubSub instead of maintaining another variable to keep track of that.

File: app/scripts/HiGlassComponent.js (hunk 2 above)
Comment (keller-mark) [reply app/scripts/HiGlassComponent.js]: It is not necessary, but it makes this event consistent with the existing mouse move event https://github.com/higlass/higlass/blob/develop/app/scripts/HiGlassComponent.js#L4044

Comment (pkerpedjiev) [reply app/scripts/HiGlassComponent.js]: Interesting. OK, I'll hold my nose and let it slide :-)

File: app/scripts/api.js (hunk 3 above)
Comment (keller-mark) [reply app/scripts/api.js]: Right now only the `publish` method is actually available in HiGlassComponent https://github.com/higlass/higlass/blob/develop/app/scripts/api.js#L32, so I will work on trying to make the listeners or number of listeners available as well

Comment (pkerpedjiev) [reply app/scripts/api.js]: OK, it's not essential but I think it would reduce a little bit of complexity.
